src/app/lib/web_channel.js

```diff
@@ -0,0 +1,39 @@
+import Events from 'ampersand-events';
+
+class WebChannel {
+  constructor() {
+    // turn this into an event emitter
+    Events.createEmitter(this);
+
+    // listen for WebChannel messages
+    window.addEventListener('WebChannelMessageToContent', this._messageReceived.bind(this));
+
+    this.channelId = null;
+  }
+
+  sendMessage(type, data) {
+    window.dispatchEvent(new window.CustomEvent('WebChannelMessageToChrome', {
+      detail: {
+        id: this.channelId,
+        message: {
+          type: type,
+          data: data
+        }
+      }
+    }));
+  }
+
+  _messageReceived(e) {
```

Comment:
style nitpick. not a huge deal, but I've always preferred to avoid 'e' and use either 'ev' or 'err'.